src/riak_kv_wm_timeseries_query.erl

```diff
@@ -54,23 +54,25 @@
 -include("riak_kv_wm_raw.hrl").
 -include("riak_kv_ts.hrl").
 
--record(ctx, {api_version,
-              method  :: atom(),
-              prefix,       %% string() - prefix for resource uris
-              timeout,      %% integer() - passed-in timeout value in ms
-              security,     %% security context
-              client,       %% riak_client() - the store client
-              riak,         %% local | {node(), atom()} - params for riak client
-              query          :: undefined | string(),
-              compiled_query :: undefined | #ddl_v1{} | #riak_sql_describe_v1{} | ?SQL_SELECT{},
-              result         :: undefined | ok | {Headers::[binary()], Rows::[ts_rec()]} |
-                                [{entry, proplists:proplist()}]
-             }).
+-record(ctx, {
+          table :: 'undefined' | binary(),
+          mod   :: 'undefined' | module(),
+          method  :: atom(),
+          prefix,       %% string() - prefix for resource uris
+          timeout,      %% integer() - passed-in timeout value in ms
+          security,     %% security context
+          riak,         %% local | {node(), atom()} - params for riak client
+          sql_type,
+          compiled_query :: undefined | #ddl_v1{} | #riak_sql_describe_v1{} | #riak_select_v1{},
+          result         :: undefined | ok | {Headers::[binary()], Rows::[ts_rec()]} |
+                            [{entry, proplists:proplist()}]
+         }).
 
 -define(DEFAULT_TIMEOUT, 60000).
 -define(TABLE_ACTIVATE_WAIT, 30).   %% wait until table's bucket type is activated
 
--define(CB_RV_SPEC, {boolean()|atom()|tuple(), #wm_reqdata{}, #ctx{}}).
+-type cb_rv_spec(T) :: {T, #wm_reqdata{}, #ctx{}}.
```

Comment:
Yes and no.

If we move it, it has to be
```Erlang
-type cb_rv_spec(T, C) :: {T, #wm_reqdat{}, C)
```
since the `#ctx{}` record is different in the modules.

Given than I am leaning towards the local definition.